sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java

```diff
@@ -204,6 +212,44 @@ public String getETag() {
         return resourceResponse.getETag();
     }
 
+    CosmosItemResponse<T> withRemappedStatusCode(
+        int statusCode,
+        double additionalRequestCharge,
+        boolean isContentResponseOnWriteEnabled) {
+
+        ResourceResponse<Document> mappedResourceResponse =
+            this.resourceResponse.withRemappedStatusCode(statusCode, additionalRequestCharge);
+
+        byte[] payload = null;
+        if (isContentResponseOnWriteEnabled) {
+            payload = this.responseBodyAsByteArray;
+        }
+
+        return new CosmosItemResponse<>(
```

Comment:
Same here, can we re-use the existing object?

Reply:
See above